node-manager/src/wallet.rs

```diff
@@ -0,0 +1,89 @@
+use std::str::FromStr;
+use futures::lock::Mutex;
+
+use bdk::blockchain::EsploraBlockchain;
+use bdk::keys::ExtendedKey;
+use bdk::template::DescriptorTemplateOut;
+use bdk::{SyncOptions, Wallet};
+use bip39::Mnemonic;
+use bitcoin::util::bip32::{ChildNumber, DerivationPath, ExtendedPrivKey};
+use bitcoin::Network;
+
+use crate::localstorage::MutinyBrowserStorage;
+
+#[derive(Debug)]
+pub struct MutinyWallet {
+    pub wallet: Mutex<Wallet<MutinyBrowserStorage>>,
+}
+
+impl MutinyWallet {
+    pub fn new(mnemonic: Mnemonic, network: Network) -> MutinyWallet {
+        let entropy = mnemonic.to_entropy();
+        let xprivkey = ExtendedPrivKey::new_master(network, &entropy).unwrap();
+        let xkey = ExtendedKey::from(xprivkey);
+        let account_number = 0;
+        let (receive_descriptor_template, change_descriptor_template) =
+            get_tr_descriptors_for_extended_key(xkey, network, account_number);
+
+        let db = MutinyBrowserStorage::default();
+
+        let wallet = Wallet::new(
+            receive_descriptor_template,
+            Some(change_descriptor_template),
+            network,
+            db,
+        )
+        .expect("Error creating wallet");
+
+        MutinyWallet {
+            wallet: Mutex::new(wallet),
```

Comment:
Does this have to be in a mutex? Every time syncing happens, which can take many seconds, every wallet actions will be blocked. Creating an address for instance isn't possible while syncing is happening. We can fix later but wanted to throw that question out there.

Reply:
A lot of the actions call `borrow_mut()` on the database which will panic if the database is being used elsewhere (ie syncing). We can change to a non-async mutex and then give an generic error message or remove it and let bdk panic instead